arrow/parser.py

```diff
@@ -230,10 +225,10 @@ def parse(self, datetime_string, fmt, normalize_whitespace=False):
 
         if match is None:
             raise ParserMatchError(
-                "Failed to match '{}' when parsing '{}'".format(fmt, datetime_string)
+                f"Failed to match '{fmt}' when parsing '{datetime_string}'"
             )
 
-        parts = {}
+        parts: Dict = {}
```

Comment:
I think parts is a dict of str -> int right? Also, we should put a TODO to maybe move this into _build_datetime since it seems to be unused in this function